pkg/clustervalidations/cluster.go

```diff
@@ -0,0 +1,28 @@
+package clustervalidations
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/clustervalidator"
+	"github.com/aws/eks-anywhere/pkg/controller"
+)
+
+// ValidateClusterReady get the CAPICluster from the client then validates that it is in a ready state.
+func ValidateClusterReady(ctx context.Context, vc clustervalidator.Config) error {
```

Comment:
Is this going to be conditionally added or is it something we need to always check?

Reply:
That depends on how the cluster validator is used. You could opt to not register this validation in a specific use case.